src/charm.py

```diff
@@ -437,9 +437,9 @@ def _prometheus_target_relation_broken(self, _):
         self._set_status()
 
     def _downstream_prometheus_scrape_relation_joined(self, _):
-        self._stored.have_prometheus = True
         if self._stored.have_nrpe:  # pyright: ignore
             self._on_nrpe_targets_changed(None)
+        self._stored.have_prometheus = True
```

Comment:
@przemeklal I suspect you were having this issue because of another failed `relation-set`. Would you be able `--replay | grep relation\-set` from the debug-log?

If that is the case, I believe this change should fix it.

Reply:
I shared logs and other details privately.